### PR TITLE
Add setting to disable harold

### PR DESCRIPTION
**Pull request by SpennyNDaJets**
## Summary
We need to allow admins access to turn off Harold for the entire team. We add a new setting for Harold that gatekeeps access for the errors and session insights features.

What this PR does:
Schema
- Adds a new setting to `AllWorkspaceSettings` to manage access to Harold in the app 
  - this setting will default `true`

Private Graph
- Create new query to fetch Workspace settings by project id for error and session page
- Fix bug in `EditWorkspaceSettings` that allows setting to be updated to false
- Add check in `EditWorkspaceSettings` to only allow admin role to update

Frontend
- Add new row for new setting
  - Must have admin role to edit setting
- Edit ToggleRow to allow tooltip over switch
- Hide Harold when new setting is disabled

## How did you test this change?
https://www.loom.com/share/68b63193cab14fcc8901a53dfeeaafc0?sid=db451b94-e58d-4edd-aaee-3da362440b7d

## Are there any deployment considerations?
N/A

### CODE REVIEW

File: backend/private-graph/graph/schema.resolvers.go [hunk 1]
```diff
@@ -622,17 +622,19 @@ func (r *mutationResolver) EditWorkspace(ctx context.Context, id int, name *stri
...
-	workspaceSettings := &model.AllWorkspaceSettings{
-		WorkspaceID: workspaceID,
+
+	workspaceSettings := &model.AllWorkspaceSettings{}
+	workspaceSettingsUpdates := map[string]interface{}{
+		"AIApplication": *aiApplication,
+		"AIInsights":    *aiInsights,
 	}
-	if err := r.DB.Where(workspaceSettings).Updates(&model.AllWorkspaceSettings{
-		AIInsights: *aiInsights,
```
Comment (SpennyNDaJets) [inline backend/private-graph/graph/schema.resolvers.go:634]: Gorm won't update field to false, so refactored to fix bug
https://stackoverflow.com/questions/56653423/gorm-doesnt-update-boolean-field-to-false

File: backend/private-graph/graph/schema.resolvers.go [hunk 2]
```diff
@@ -6766,7 +6787,7 @@ func (r *queryResolver) SubscriptionDetails(ctx context.Context, workspaceID int
 	}
 
 	if err := r.validateAdminRole(ctx, workspaceID); err != nil {
-		return nil, nil
+		return nil, err
```
Comment (SpennyNDaJets) [inline backend/private-graph/graph/schema.resolvers.go:6790]: Just caught this, but unrelated

**Review by ccschmitz — APPROVED** [review]
Looks great, @SpennyNDaJets - nice work! Had one question about adding a new query for fetching workspace settings, but that's minor, and I'm probably missing something about why it's needed 😸

File: backend/private-graph/graph/schema.resolvers.go [hunk 3]
```diff
@@ -6539,6 +6545,21 @@ func (r *queryResolver) WorkspaceSettings(ctx context.Context, workspaceID int)
...
+	}
+
+	workspaceSettings := model.AllWorkspaceSettings{}
+	if err := r.DB.Where(model.AllWorkspaceSettings{WorkspaceID: project.WorkspaceID}).First(&workspaceSettings).Error; err != nil {
+		return nil, err
+	}
+
+	return &workspaceSettings, nil
+}
```
[the remark this answers is not in the document]
Comment (SpennyNDaJets) [reply backend/private-graph/graph/schema.resolvers.go:6561]: Maybe thats my understanding, but was struggling to find the workspace id. I saw the project_id in the url, but is there a good way to find the workspace id?

[the remark this answers is not in the document]
Comment (ccschmitz) [reply backend/private-graph/graph/schema.resolvers.go:6561]: Yeah, not obvious where to find it, but I think it's available inside `ApplicationContext`:

https://github.com/highlight/highlight/blob/02a38aae05971bc272e6853d110a0af1dcc14106/frontend/src/routers/AppRouter/AppRouter.tsx#L193-L201

[the remark this answers is not in the document]
Comment (SpennyNDaJets) [reply backend/private-graph/graph/schema.resolvers.go]: Thanks for catching this! Just updated with to use the new context and clean up the unnecessary query

**Pull request merged**